keps/sig-network/2200-externalips-admission/README.md

```diff
@@ -0,0 +1,257 @@
+  It could start disallowing all Service operations, if the controller was
+  buggy.
+
+* **What specific metrics should inform a rollback?**
+  `apiserver_admission_controller_admission_duration_seconds_bucket{name="DenyServiceExternalIPs",rejected="true", ...}`
+`
+
+* **Were upgrade and rollback tested? Was the upgrade->downgrade->upgrade path tested?**
+  TODO
+  Describe manual testing that was done and the outcomes.
+  Longer term, we may want to require automated upgrade/rollback tests, but we
+  are missing a bunch of machinery and tooling and can't do that now.
+
+* **Is the rollout accompanied by any deprecations and/or removals of features, APIs, 
+fields of API types, flags, etc.?**
+  No.
+
+### Monitoring Requirements
+
+* **How can an operator determine if the feature is in use by workloads?**
+  Via kubectl:
+  ```
+  kubectl get svc --all-namespaces -o go-template='
+  {{- range .items -}}
+    {{if .spec.externalIPs -}}
+      {{.metadata.namespace}}/{{.metadata.name}}: {{.spec.externalIPs}}{{"\n"}}
+    {{- end}}
+  {{- end -}}
+  '
+  ```
+
+* **What are the SLIs (Service Level Indicators) an operator can use to determine 
+the health of the service?**
+  N/A
+
+* **What are the reasonable SLOs (Service Level Objectives) for the above SLIs?**
+  N/A
+
+* **Are there any missing metrics that would be useful to have to improve observability 
+of this feature?**
+  This proposes to use the existing
+  `apiserver_admission_controller_admission_duration_seconds_bucket{name="DenyServiceExternalIPs", ...}` metrics.
+
+### Dependencies
+
+* **Does this feature depend on any specific services running in the cluster?**
+  No.
+
+### Scalability
+
+* **Will enabling / using this feature result in any new API calls?**
+ No.
+
+* **Will enabling / using this feature result in introducing new API types?**
+ No.
+
+* **Will enabling / using this feature result in any new calls to the cloud 
+provider?**
+ No.
+
+* **Will enabling / using this feature result in increasing size or count of 
+the existing API objects?**
+ No.
+
+* **Will enabling / using this feature result in increasing time taken by any 
+operations covered by [existing SLIs/SLOs]?**
+ No.
+
+* **Will enabling / using this feature result in non-negligible increase of 
+resource usage (CPU, RAM, disk, IO, ...) in any components?**
+ No.
+
+### Troubleshooting
+
+* **How does this feature react if the API server and/or etcd is unavailable?**
+  It is part of apiserver REST path.
+
+* **What are other known failure modes?**
+  None.
+
+* **What steps should be taken if SLOs are not being met to determine the problem?**
+  N/A
+
+## Implementation History
+
+* 2020-12-07: First draft
+
+## Drawbacks
+
+It is a slippery-slope to other ad hoc policies.  Counter: this is very
+surgical and overwhelmingly not a useful feature.
+
+## Alternatives
+
+* Force users to use policy controllers as webhooks. Forever.
+* Make a breaking API change and disable or rip-out the feature.
```

Comment:
Long term, I would prefer us to make the breaking API change, but given the sensitivity of breaking API changes, this KEP is a lovely short term solution to make clusters safer and give the time necessary to break the API without being in a rush.